src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/ProjectKey.cs

```diff
@@ -19,7 +18,7 @@ namespace Microsoft.CodeAnalysis.Razor.ProjectSystem;
     // end up. All creation logic is here in one place to ensure this is consistent.
     public static ProjectKey From(HostProject hostProject) => new(hostProject.IntermediateOutputPath);
     public static ProjectKey From(IProjectSnapshot project) => new(project.IntermediateOutputPath);
-    public static ProjectKey? From(Project project)
+    public static ProjectKey From(Project project)
```

Comment:
>Project [](http://example.com/codeflow?start=34&length=7)

Dumb question, could the Project/HostProject/IProjectSnapshot be kept in a field and do a ReferenceEquals check before calling AreDirectoryPathsEquivalent?

Reply:
I'm guessing the answer is that it wouldn't hit often enough to be worth it, which makes sense in looking at RazorDynamicFileInfoProvider.TryFindProjectIdForProjectKey

Reply:
Nope -- and these keys are supposed to be valid across snapshots as well. Think of them as our version of `ProjectId` that works for both Roslyn projects _and_ Razor project snapshots.